source/extensions/tracers/datadog/agent_http_client.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <datadog/http_client.h>
+
+#include <string>
+
+#include "envoy/http/async_client.h"
+
+#include "source/common/common/logger.h"
+#include "source/common/upstream/cluster_update_tracker.h"
+#include "source/extensions/tracers/datadog/dd.h"
+
+#include "absl/container/flat_hash_map.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+
+struct TracerStats;
+
+class AgentHTTPClient : public dd::HTTPClient,
+                        public Http::AsyncClient::Callbacks,
+                        private Logger::Loggable<Logger::Id::tracing> {
+public:
+  struct Handlers {
+    ResponseHandler on_response;
+    ErrorHandler on_error;
+  };
+
+private:
+  absl::flat_hash_map<Http::AsyncClient::Request*, Handlers> handlers_;
+  Upstream::ClusterUpdateTracker collector_cluster_;
+  std::string cluster_;
+  std::string reference_host_;
+  TracerStats* stats_;
```

Comment:
use a reference here, or if this is optional, then an OptRef.

Reply:
:heavy_check_mark: